entproto/service.go

```diff
@@ -25,33 +25,54 @@ import (
 	_ "google.golang.org/protobuf/types/known/emptypb"
 )
 
-type method string
-
 const (
 	ServiceAnnotation        = "ProtoService"
-	get               method = "Get"
-	create                   = "Create"
-	update                   = "Update"
-	delete_                  = "Delete"
+	MethodCreate      Method = 1 << iota
+	MethodGet
+	MethodUpdate
+	MethodDelete
+	MethodAll = MethodCreate | MethodGet | MethodUpdate | MethodDelete
 )
 
 var (
 	errNoServiceDef = errors.New("entproto: annotation entproto.Service missing")
 )
 
+type Method uint
+
+// Is reports whether method m matches given method n.
+func (m Method) Is(n Method) bool { return m&n != 0 }
+
+func Methods(methods Method) Method {
+	return methods
+}
+
 type service struct {
 	Generate bool
+	Methods  Method
 }
 
 func (service) Name() string {
 	return ServiceAnnotation
 }
 
-func Service() schema.Annotation {
-	return service{Generate: true}
+func Service(methods ...Method) schema.Annotation {
```

Comment:
should be a functional option. think that we may want other parameters. in other orders
`entproto.Service(entproto.MethodAll)` should not work. 

if you're not familiar with the concept see this post its pretty cool:
https://dave.cheney.net/2014/10/17/functional-options-for-friendly-apis

and see the implementation in a file right next to you: https://github.com/ent/contrib/blob/master/entproto/message.go#L28-L40

Reply:
Thanks! I'll work on incorporating this into my PR. I also didn't realize I could do bit shifts with iota. Very cool!

If it wasn't clear, I'm still relatively new to Go; I'm still learning some of the design patterns for it.

Reply:
You're doing great! Thanks for being patient and pushing this through!

Reply:
Okay, I think I've implemented this design pattern appropriately now. It is definitely quite neat!